### DISCUSSION TITLE
MSBuild location on Mono 5+ is not set correctly on Linux and Mac OS X

### DISCUSSION THREAD

**Discussion by PawelTroka** [cake-build/cake#3527] [Q&A]
Cake v0.19.5, Travis CI.

Log from Linux:

> Executing task: Build
> An error occurred when executing task 'Build'.
> Error: Path cannot be empty.
> Parameter name: path

Build: https://travis-ci.org/PawelTroka/Computator.NET/jobs/239292355

Log from Mac OS X:

> An error occurred when executing task 'Build'.
> Error: MSBuild: Could not locate executable.

Build: https://travis-ci.org/PawelTroka/Computator.NET/jobs/239737557

Also nuget restore is still using xbuild in case of Linux and is already using MSBuild in case of Mac OS X.
Linux:

> Executing task: Restore
> sbuild version '14.0' from '/usr/lib/mono/xbuild/14.0/bin'.

Mac OS X:

> Executing task: Restore
> sbuild version '15.0' from '/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/msbuild/15.0/bin/'.

Workaround code I had written to make it work correctly (in case of builds - not fixing nuget using xbuild on Linux):
```
	if(!IsRunningOnWindows() && isMonoButSupportsMsBuild)
	{
		msBuildSettings.ToolPath = new FilePath(
		  System.Environment.OSVersion.Platform != System.PlatformID.MacOSX
		  ? @"/usr/lib/mono/msbuild/15.0/bin/MSBuild.dll"
		  : @"/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/msbuild/15.0/bin/MSBuild.exe"
		  );//hack for Linux and Mac OS X bug - missing MSBuild path	
	}
```

**Comment by JeremyCade** [comment]
Can you set `FrameworkPathOverride`? 

e.g. 
```
export FrameworkPathOverride=/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/4.5/
```
